rskj-core/src/main/java/co/rsk/scoring/InetAddressCidrBlock.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2022 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.scoring;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.net.InetAddress;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * InetAddressBlock represents a range of InetAddress
+ * it has a number of bits to ignore
+ * <p>
+ * Created by ajlopez on 11/07/2017.
+ */
+public class InetAddressCidrBlock {
+    private final String description;
+    private final byte[] bytes;
+
+    // new algorithm data
+    private final int mask;
+    private final int maskBytesCount;
+
+    /**
+     * Creates an InetAddressBlock given an address and the number of bits to ignore
+     *
+     * @param address the address
+     * @param bits    the numbers of bits to ignore
+     */
+    public InetAddressCidrBlock(InetAddress address, int bits) {

```

Comment:
this ctor is public one.. can we validate `bits` argument here? There's already some validation in `co.rsk.scoring.InetAddressUtils#parse`, but an instance can be created directly by this ctor.

Reply:
Good point! I added the suggested validation and also made it package-protected as it was only used from `InetAddressUtils#parse` and `InetAddressCidrBlockTest` tests